File: BOJ/2028.cpp
#include <cstdio>
#include <algorithm>
#include <cstring>
#include <math.h>
typedef long long ll;
int t;
int main(){
	scanf("%d", &t);
	while (t--) {
		ll n;
		scanf("%lld", &n);
		ll temp = n*n;
		ll sum = 0;
		int cnt = 0;
		bool ok = false;
		while (temp > 0){
			sum += (temp % 10) * pow(10, cnt);
			if (sum == n) {
				ok = true;
				break;
			}
			temp /= 10;
			cnt++;
		}
		if (ok) printf("YES\n");
		else printf("NO\n");
	}
	return 0;
}

File: BOJ(Python 공부용)/2028.py
import sys
from heapq import heappush, heappop
from math import sqrt

input = sys.stdin

if __name__ == '__main__':
    t = int(input.readline())
    for _ in range(t):
        n = int(input.readline())
        temp = n * n
        sum, cnt = 0, 0
        ok = False

        while temp > 0:
            sum += int((temp % 10) * pow(10, cnt))
            if sum == n:
                ok = True
                break
            temp //= 10
            cnt += 1
        if ok:
            print("YES")
        else:
            print("NO")
